siddhant/random-wallpaper-wallhaven.py

```diff
@@ -0,0 +1,31 @@
+import requests, bs4, os, random
+#import subprocess
+url = "https://alpha.wallhaven.cc/toplist"
+dir_name = 'wallpapers-wallhaven-toplist'
+if not os.path.exists(dir_name):
+    os.makedirs(dir_name)
+print("Downloading Page %s......" % url)
+res = requests.get(url)
+res.raise_for_status()
+soup = bs4.BeautifulSoup(res.text, "html.parser")
+top_images_list = soup.select('section ul li figure a')
+if top_images_list == []:
+    print('Could not find images!')
+else:
+    random_image_url = top_images_list[random.randint(1, len(top_images_list))].get('href')
+    res = requests.get(random_image_url)
+    res.raise_for_status()
+    soup = bs4.BeautifulSoup(res.text, 'html.parser')
+    image = soup.select('main section div img')
+    image_url = 'https:' + image[0].get('src')
+    print('Downloading image.... %s' % image_url)
+    res = requests.get(image_url)
+    res.raise_for_status()
+    imageFile = open(os.path.join('wallpapers-wallhaven-toplist', os.path.basename(image_url)), 'wb')
```

Comment:
Try to use a relative path so that the script can be run from anywhere and still it works.

Reply:
The script creates the folder 'wallpapers-wallhaven-toplist' wherever it is run and the image is downloaded in that folder. So path independent.

Reply:
Sounds good then! :+1: